bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/CPU.java

```diff
@@ -2113,6 +2269,11 @@ private static void execTypeConversionOpcodes(WorkerExecutionContext ctx, Worker
                 j = operands[1];
                 sf.intRegs[j] = sf.longRegs[i] != 0 ? 1 : 0;
                 break;
+            case InstructionCodes.I2D:
+                i = operands[0];
+                j = operands[1];
+                sf.refRegs[j] = new BDecimal(new BInteger(sf.longRegs[i]).decimalValue());
```

Comment:
shall we create a new BigDecimal here without wrapping the value in a BInteger?

Reply:
Addressed with commit ce04700.